### PR TITLE
Feature map

### PR DESCRIPTION
**Pull request by LaraMo**

### CODE REVIEW

File: src/MainDriver.cpp [hunk 1]
```diff
@@ -1,11 +1,13 @@
 #include "OrdersList/Orders.h"
 #include "Player/Player.h"
 #include "CardsDeck/Cards.h"
+#include "Map/Map.h"
 
 int main()
 {
-    testOrdersLists();
+    /*testOrdersLists();
```
Comment (LaraMo) [inline src/MainDriver.cpp:8]: This is temporary, don't wanna flood the terminal with other logs!

**Comment by LaraMo** [comment]
This merge includes the following changes:
- [ ]  Adding a param constructor for Map
- [ ]  Adding and testing a copy constuctor for Map
- [ ]  Reading 2 types of `.map` files and parsing them into respective Continents and Territories
- [ ] - Create adjacency for all territories
- [ ] Print the map

TODO:
- [ ] Add more map files to we can test to find any bugs
- [ ] Fix continents not printing for solar system map file
- [ ] Create a validation method (3 conditions must be validated)
- [ ] Test invalid map loading, for me maps still load even if invalid, wierd!!!!
- [ ] Delete all pointers created by the Map

File: src/MainDriver.cpp (hunk 1 above)
Comment (EyalAzimov) [reply src/MainDriver.cpp:8]: Good call!

File: src/Player/PlayerDriver.cpp [hunk 2]
```diff
@@ -10,12 +10,12 @@ void testPlayers()
...
-  Territory* t3 = new Territory("Russia");
-  Territory* t4 = new Territory("Israel");
-  Territory* t5 = new Territory("Italy");
-  Territory* t6 = new Territory("Bulgaria");
+  Territory* t1 = new Territory("UK", 1, 2);
+  Territory* t2 = new Territory("USA", 2, 3);
+  Territory* t3 = new Territory("Russia", 3, 3);
+  Territory* t4 = new Territory("Israel", 4, 2);
+  Territory* t5 = new Territory("Italy", 5, 3);
+  Territory* t6 = new Territory("Bulgaria", 6, 3);
```
[the remark this answers is not in the document]
Comment (EyalAzimov) [reply src/Player/PlayerDriver.cpp:18]: Looks good and intuitive! Thanks!

File: README.md [hunk 3]
```diff
@@ -1,19 +1,28 @@
...
+  "code-runner.executorMap": {
+
+        "javascript": "node",
+        "java": "cd $dir && javac $fileName && java $fileNameWithoutExt",
+        "c": "cd $dir && gcc $fileName -o $fileNameWithoutExt && $dir$fileNameWithoutExt",
+        "zig": "zig run",
+        "cpp": "cd $dir && g++ -std=c++14 $fileName `find . \\( -iname '*.cpp' -not -name '$fileName' \\)` -o $fileNameWithoutExt && $dir$fileNameWithoutExt",
+....
+}
+```
```
Comment (EyalAzimov) [inline README.md:28]: Easier to undserstand!! I'll this style next time!

File: src/Map/Map.h [hunk 4]
```diff
@@ -0,0 +1,129 @@
...
+class Continent {
+private:
+    int id;
+    string name;
+
+public:
+
+    /// <summary>
+    /// Default Constructor
+    /// </summary>
```
Comment (EyalAzimov) [inline src/Map/Map.h:21]: Can we get rid of these comments in the future? And just add them at end of the same line if necessary, instead of on top?

**Review by EyalAzimov — APPROVED** [review]
Great job!!
Everything ran smoothly on my end!
I didn't read all of it, but I'll look through more during the meeting!

File: src/Map/Map.h (hunk 4 above)
Comment (LaraMo) [reply src/Map/Map.h:21]: image
after reading, this is also a good standard if you prefer to comment the methods like so, let me know:)

**Pull request merged**